lib/mongoid/reloadable.rb

```diff
@@ -22,7 +22,7 @@ def reload
       end
 
       reloaded = _reload
-      if Mongoid.raise_not_found_error && (!reloaded || reloaded.empty?)
+      if Mongoid.raise_not_found_error && reloaded.empty?
```

Comment:
The docstring on `_reload` should also then be updated to say that if the embedded document no longer exists in the dabatase, `_reload` would return nil.

Subsequently, here I think the correct fix is to check `reloaded.nil?` either instead of or in addition to `reloaded.empty?`.

I am also curious what it means for `reloaded` to be empty here - the document exists but has no attributes? How does a user attain such a beast?

Reply:
Yeah I'm not exactly sure how this case is achieved, but in general I balk at removing safety checks.

Reply:
I checked this situation I am not a fan of how that is implemented either, let's leave it alone for now since it's not in scope of your work.